### PR TITLE
[IOS-2980]Fix for sequential download stuck issue

### PR DESCRIPTION
**Pull request by YueVungle**
This commit will add fix for sequential download stuck.

IOS-2980

### CODE REVIEW

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m [hunk 1]
```diff
@@ -269,7 +273,11 @@ - (nullable NSError *)loadAd:(nonnull NSString *)placement
   if ([self isAdCachedForPlacementID:placement withDelegate:delegate]) {
     [delegate adAvailable];
   } else {
-    NSError *loadError;
+    if ([_prioritizedPlacementID isEqualToString:placement]) {
+      return nil;
+    }
```
Comment (shiv-vungle) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:278]: Can we add a comment why we are returning nil if placement is same as previous placement?

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m [hunk 2]
```diff
@@ -101,6 +104,7 @@ - (void)initWithAppId:(nonnull NSString *)appId
   if (delegate) {
     NSString *priorityPlacementID = delegate.desiredPlacement;
     [initOptions setObject:priorityPlacementID forKey:VungleSDKInitOptionKeyPriorityPlacementID];
+    _prioritizedPlacementID = [priorityPlacementID copy];
```
Comment (ashinagawa) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:107]: I wonder if we need to create and add "delegate" at this point before calling init (which leads to requesting ad)?
Just curious.
I run the project and verified it's working without problem, so I approved this for now for your to build timely.
If you need to change it, I will review again. Please let me know.

Comment (YueVungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:107]: @ashinagawa Hi Aki, thanks a lot for the reminder, in the top of this method `initWithAppId:delegate:`, we already added the "delegate":

 if (delegate) {
    GADMAdapterVungleMapTableSetObjectForKey(_initializingDelegates, delegate.desiredPlacement,
                                             delegate);
  }

^_^

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m (hunk 1 above)
Comment (YueVungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:278]: @shiv-vungle Got it, I will add a comment to explain why we return nil in this case. Thanks a lot for the reminder. ^_^

Comment (YueVungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:278]: @shiv-vungle Added the comment to explain why we return nil in this case in `6.7.x_pre_release` branch. Thanks a lot. ^_^

**Pull request merged**